docs/_spec/03-types.md

```diff
@@ -492,20 +491,76 @@ union : [A >: Nothing <: Comparable[A]] (x: Set[A], xs: Set[A]) Set[A]
 
 ### Type Constructors
 
-A _type constructor_ is represented internally much like a polymorphic method type.
-`[´\pm´ ´a_1´ >: ´L_1´ <: ´U_1, ..., \pm a_n´ >: ´L_n´ <: ´U_n´] ´T´` represents a type that is expected by a [type constructor parameter](04-basic-declarations-and-definitions.html#type-parameters) or an [abstract type constructor binding](04-basic-declarations-and-definitions.html#type-declarations-and-type-aliases) with the corresponding type parameter clause.
+```
+Type            ::=  ... |  HkTypeParamClause ‘=>>’ Type
+TypeParamClause ::=  ‘[’ TypeParam {‘,’ TypeParam} ‘]’
+TypeParam       ::=  {Annotation} (id [HkTypeParamClause] | ‘_’) TypeBounds
+TypeBounds      ::=  [‘>:’ Type] [‘<:’ Type]
```

Comment:
`HkTypeParamClause` does not seem to exist in this spec, and `TypeParamClause` conflicts with it somewhere else where variance annotations are allowed. I have a syntax suggestion at https://github.com/sjrd/dotty/commit/84cf279543341693bb1345ffce68566938c1836a.

(Also, the full syntax of types is at the top of the file, not spread in each section.)